exchanger/src/main/kotlin/com/d3/exchange/exchanger/ExchangerService.kt

```
@@ -85,24 +90,53 @@ class ExchangerService(
     /**
      * Indicates if there was an update of the trade pairs value and loads it
      */
-    private fun updateTradingPairs(transaction: TransactionOuterClass.Transaction) {
+    private fun updateTradingPairsOnBlock(transaction: TransactionOuterClass.Transaction) {
         val reducedPayload = transaction.payload.reducedPayload
-        if (reducedPayload.creatorAccountId == tradePairSetter &&
-            reducedPayload.commandsList.any { command ->
+        if (reducedPayload.creatorAccountId == tradePairSetter) {
+            val exchangerDetails = reducedPayload.commandsList.filter { command ->
                 command.hasSetAccountDetail()
                         && command.setAccountDetail.accountId == exchangerAccountId
-                        && command.setAccountDetail.key == tradePairKey
+            }.map {
+                it.setAccountDetail
+            }
+            if (exchangerDetails.any { command -> command.key == tradePairKey }) {
+                updateTradingPairs()
             }
-        ) {
-            queryhelper.getAccountDetails(exchangerAccountId, tradePairSetter, tradePairKey).map {
-                if (it.isPresent) {
-                    val unEscape = it.get().irohaUnEscape()
-                    tradingPairs = gson.fromJson<Map<String, Set<String>>>(
-                        unEscape,
-                        typeToken
-                    )
-                    logger.info { "Updated pairs: $unEscape" }
-                }
+            if (exchangerDetails.any { command -> command.key == unusualAssetsKey }) {
+                updateUnusualAssets()
+            }
+        }
+    }
+
+
+    /**
+     * Queries Iroha for trading pairs details and loads it
+     */
+    private fun updateTradingPairs() {
+        queryhelper.getAccountDetails(exchangerAccountId, tradePairSetter, tradePairKey).map {
```

Errors are not handled